File: 프로그래머스/Level1/로또의 최고 순위와 최저 순위.py
# https://programmers.co.kr/learn/courses/30/lessons/77484

def solution(lottos, win_nums):
    cnt = 0
    zero_cnt = lottos.count(0)
    for num in lottos:
        if num in win_nums:
            cnt += 1
    result = [6,6,5,4,3,2,1]
    answer = [result[cnt+zero_cnt], result[cnt]]
    return answer

File: 프로그래머스/Level1/로또의 최고 순위와 최저 순위.js
// https://programmers.co.kr/learn/courses/30/lessons/77484

function solution(lottos, win_nums) {
  const rank = [6, 6, 5, 4, 3, 2, 1];
  const cnt = lottos.reduce((prev, cur) => (win_nums.includes(cur) ? prev + 1 : prev), 0);
  const zero = lottos.reduce((prev, cur) => (cur === 0 ? prev + 1 : prev), 0);
  return [rank[cnt + zero], rank[cnt]];
}
